docs/requirements.txt

```diff
@@ -1,8 +1,8 @@
 #
-# This file is autogenerated by pip-compile with Python 3.9
+# This file is autogenerated by pip-compile with Python 3.11
 # by the following command:
 #
-#    pip-compile --allow-unsafe --generate-hashes --output-file=docs/requirements.txt --strip-extras docs/requirements.in
+#    pip-compile --allow-unsafe --config=pyproject.toml --generate-hashes --output-file=docs/requirements.txt --strip-extras docs/requirements.in
```

Comment:
The deps aren't coming from that config, no need to add it here.

Reply:
This is being added by pip-compile, I am not adding it myself. That is the line generated from running `pip-compile --upgrade-package pyyaml --generate-hashes --strip-extras --output-file=docs/requirements.txt --allow-unsafe docs/requirements.in`.

Reply:
@Qalthos oh, I think I _know_ why this is happening. Recently, jag added support for configs to pip-tools: https://github.com/jazzband/pip-tools/pull/1863. I've performed a series of in-depth reviews and got it merged eventually. But afterward, a few shortcomings have been noticed and addressed (I think some of those are still in progress).
So the related bug report is https://github.com/jazzband/pip-tools/issues/1903 and the pending bugfix PR is https://github.com/jazzband/pip-tools/pull/1904.

Also, I somehow missed that it was a `--config` rather than a dependencies source, which confused me originally. With that in mind, I'm going to mark this thread as resolved.